controllers/effective_dnspolicies_reconciler.go

```diff
@@ -300,3 +297,30 @@ func (r *EffectiveDNSPoliciesReconciler) deleteRecord(ctx context.Context, obj m
 		logger.Error(err, "failed to delete DNSRecord", "record", obj.GetLocator())
 	}
 }
+
+// canUpdateDNSRecord returns true if the current record can be updated to the desired.
+func canUpdateDNSRecord(ctx context.Context, current, desired *kuadrantdnsv1alpha1.DNSRecord) bool {
+	logger := controller.LoggerFromContext(ctx)
+
+	// DNSRecord doesn't currently support rootHost changes
+	if current.Spec.RootHost != desired.Spec.RootHost {
+		logger.V(1).Info("root host for existing record has changed")
+		return false
+	}
+
+	// DNSRecord doesn't currently support record type changes due to a limitation of the dns operator
+	// https://github.com/Kuadrant/dns-operator/issues/287
+	for _, curEp := range current.Spec.Endpoints {
+		for _, desEp := range desired.Spec.Endpoints {
+			if curEp.DNSName == desEp.DNSName {
+				if curEp.RecordType != desEp.RecordType {
+					logger.V(1).Info("record type for existing endpoint has changed",
+						"dnsName", curEp.DNSName, "current", curEp.RecordType, "desired", desEp.RecordType)
+					return false
+				}
+			}
+		}
+	}
+
+	return true
+}
```

Comment:
@maleck13 fyi, because i had to do this to deal with possible changes in record type when attaching a policy to a listener that already has existing records via policy attached to a gateway, it means we could probably remove this now https://github.com/Kuadrant/kuadrant-operator/blob/main/api/v1alpha1/dnspolicy_types.go#L56 and allow switching from simple to loadbalanced without the need for manual intervention.

Reply:
nice ok. Lets go for it

Reply:
I'll do it in a different PR, will need some tests

Reply:
https://github.com/Kuadrant/kuadrant-operator/pull/968